sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/aad/properties/AadAuthenticationProperties.java

```diff
@@ -563,77 +566,29 @@ private void validateAuthorizationClientProperties(String registrationId,
         } else {
             // Validate authorization grant grantType
-            switch (applicationType) {
-                case WEB_APPLICATION:
-                    if (ON_BEHALF_OF.getValue().equals(grantType)) {
-                        throw new IllegalStateException(String.format(UNMATCHING_OAUTH_GRANT_TYPE_FROMAT,
-                            "web_application", registrationId, "on_behalf_of"));
-                    }
-                    break;
-                case RESOURCE_SERVER:
-                    if (AUTHORIZATION_CODE.getValue().equals(grantType)) {
-                        throw new IllegalStateException(String.format(UNMATCHING_OAUTH_GRANT_TYPE_FROMAT,
-                            "resource_server", registrationId, "authorization_code"));
-                    }
-                    if (ON_BEHALF_OF.getValue().equals(grantType)) {
-                        throw new IllegalStateException(String.format(UNMATCHING_OAUTH_GRANT_TYPE_FROMAT,
-                            "resource_server", registrationId, "on_behalf_of"));
-                    }
-                    break;
-                case RESOURCE_SERVER_WITH_OBO:
-                    if (AUTHORIZATION_CODE.getValue().equals(grantType)) {
-                        throw new IllegalStateException(String.format(UNMATCHING_OAUTH_GRANT_TYPE_FROMAT,
-                            "resource_server_with_obo", registrationId, "authorization_code"));
-                    }
-                    break;
-                case WEB_APPLICATION_AND_RESOURCE_SERVER:
-                default:
-                    LOGGER.debug("'spring.cloud.azure.active-directory.authorization-clients." + registrationId
-                        + ".authorization-grant-type' is valid.");
-            }
-
-            if (AZURE_CLIENT_REGISTRATION_ID.equals(registrationId)
-                && AUTHORIZATION_CODE != properties.getAuthorizationGrantType()) {
-                throw new IllegalStateException("spring.cloud.azure.active-directory.authorization-clients."
-                    + AZURE_CLIENT_REGISTRATION_ID
-                    + ".authorization-grant-type must be configured to 'authorization_code'.");
-            }
+            validateAuthorizationGrantType(registrationId, grantType, properties);
         }
 
         // Validate scopes.
-        List<String> scopes = properties.getScopes();
-        if (scopes == null || scopes.isEmpty()) {
-            throw new IllegalStateException(
-                "'spring.cloud.azure.active-directory.authorization-clients." + registrationId + ".scopes' must be "
-                    + "configured");
-        }
-        // Add necessary scopes for authorization_code clients.
-        // https://docs.microsoft.com/graph/permissions-reference#remarks-17
-        // https://docs.microsoft.com/azure/active-directory/develop/v2-permissions-and-consent#openid-connect-scopes
+        List<String> scopes = validateScopes(registrationId, properties);
```

Comment:
No need to extract a method for this validating scopes.

Reply:
why?

Reply:
The validateScopes couples two things: extracting the scopes and validating them. To me the extracting shouldn't be in the validating method, then if we move the extraction out of the method, then there's no need to have a separate method for validating it, for it's just one line code.

Reply:
extractValidatedScopes will be used. I think this is better.